File: src/main/java/nz/co/searchwellington/commentfeeds/detectors/WellingtonScoopCommentFeedDetector.java
package nz.co.searchwellington.commentfeeds.detectors;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class WellingtonScoopCommentFeedDetector implements CommentFeedDetector {

	static Pattern commentFeedUrlPattern = Pattern.compile("^http://wellington.scoop.co.nz/\\?feed=rss2\\&p=(\\d+)$");
	static Pattern httpsCommentFeedUrlPattern = Pattern.compile("^https://wellington.scoop.co.nz/\\?feed=rss2\\&p=(\\d+)$");

	public boolean isValid(String url) {
		Matcher matcher = commentFeedUrlPattern.matcher(url);
		Matcher httpsMatcher = httpsCommentFeedUrlPattern.matcher(url);
		return url != null && (matcher.matches() || httpsMatcher.matches());
	}
	
}

File: src/main/scala/nz/co/searchwellington/commentfeeds/detectors/WellingtonScoopCommentFeedDetector.scala
package nz.co.searchwellington.commentfeeds.detectors

import org.springframework.stereotype.Component

import java.net.URL
import java.util.regex.Pattern

@Component
class WellingtonScoopCommentFeedDetector extends CommentFeedDetector {

  private val commentFeedUrlPattern = {
    Pattern.compile("""^http://wellington.scoop.co.nz/\?feed=rss2\&p=(\d+)$""")
  }
  private val httpsCommentFeedUrlPattern = {
    Pattern.compile("""^https://wellington.scoop.co.nz/\?feed=rss2\&p=(\d+)$""")
  }

  override def isValid(url: URL): Boolean = {
    commentFeedUrlPattern.matcher(url.toString).matches || httpsCommentFeedUrlPattern.matcher(url.toExternalForm).matches
  }
}